src/Workspaces/Core/Portable/Remote/RemoteHostSessions.cs

```diff
@@ -0,0 +1,280 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Execution;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    /// <summary>
+    /// Helper type of common session case
+    /// 
+    /// RemoteHostClient.Session with 1 solution
+    /// </summary>
+    internal sealed class SessionWithSolution : IDisposable
```

Comment:
Oh.  A SessionWithSolution isn't a Session itself?  That's very confusing.  Why is this a containment design, and not an inheritance design?